ixwebsocket/IXWebSocketTransport.cpp

```diff
@@ -328,12 +336,14 @@ namespace ix
             _socket->close();
         }
 
-        // Avoid a race condition where we get stuck in select
-        // while closing.
-        if (_readyState == CLOSING)
+        if (_readyState == CLOSING && closingDelayExceeded())
         {
+            // close code and reason were set when calling close()
             _socket->close();
```

Comment:
We should call _rxbuf.clear() here too, unless we have a chance to do it later in closeSocketAndSwitchToClosedState

Reply:
I will see where it is better to add it